jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java

```diff
@@ -688,7 +688,7 @@ public Response payAllInvoices(@PathParam("accountId") final String accountId,
                                            invoice.getBalance() : remainingRequestPayment;
             if (amountToPay.compareTo(BigDecimal.ZERO) > 0) {
                 final UUID paymentMethodId = externalPayment ? null : account.getPaymentMethodId();
-                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, pluginProperties, callContext);
+                createPurchaseForInvoice(account, invoice.getId(), amountToPay, paymentMethodId, externalPayment, UUIDs.randomUUID().toString(), UUIDs.randomUUID().toString(), pluginProperties, callContext);
```

Comment:
Which test is failing? What kind of exception do you get?

The pattern is different for `PaymentAttemptModelDao`: the object refers to a transaction, it doesn't _own_ `transactionId` / `transactionExternalKey`. Besides, setting it there would make the payment control context out-of-sync (unless you update it), because the transaction key would still be `null`.

That being said, I now remember why the inconsistency was hard to fix: the id / key are created by `PaymentAutomatonDAOHelper`, called by `PaymentLeavingStateCallback` (payment state machine). At this point, the `priorCall` from the payment control state machine was already invoked (when we need the key). So I would suggest not to change it, and leave `DefaultControlInitiated` as-is (still pass `null` for both keys from JAX-RS though).

Also, FYI, it looks like you'll need to re-base the PR (GitHub reports conflicts).